### ISSUE TITLE
Is there a way to print the accuracy after each epoch?

### ISSUE THREAD

**Issue by bhushaan** [matterport/Mask_RCNN#751]
Currently it only prints losses but I was wondering if it is possible to print accuracy too.

**Comment by darrell-rg** [comment]
make a class that extends tf.keras.callbacks.Callback.on_epoch_end and add it to the list of callbacks in model.py

Example:

```
class FloydMetrics(tf.keras.callbacks.Callback):
    def on_epoch_end(self, epoch, logs={}):
        lr = K.get_value(self.model.optimizer.lr) 
        print('{{"metric": "lr", "value": {}}}'.format(lr))
        for k, v in logs.items():
            print('{{"metric": "{}", "value": {}}}'.format(k,v))
```

**Comment by bhushaan** [comment]
@darrell-rg : Where is it calculated in model ? I want the individual accuracies associated for each loss.  I am currently calculating the rpn_class_accuracy and mcrnn_class_accuracy as follows:

```
def rpn_class_accuracy_graph(rpn_match, rpn_class_logits):
    """RPN anchor classifier accuracy.

    rpn_match: [batch, anchors, 1]. Anchor match type. 1=positive,
               -1=negative, 0=neutral anchor.
    rpn_class_logits: [batch, anchors, 2]. RPN classifier logits for FG/BG.
    """
    # Squeeze last dim to simplify
    rpn_match = tf.squeeze(rpn_match, -1)
    # Get anchor classes. Convert the -1/+1 match to 0/1 values.
    anchor_class = K.cast(K.equal(rpn_match, 1), tf.int32)
    # Positive and Negative anchors contribute to the loss,
    # but neutral anchors (match value = 0) don't.
    indices = tf.where(K.not_equal(rpn_match, 0))
    # Pick rows that contribute to the loss and filter out the rest.
    rpn_class_logits = tf.gather_nd(rpn_class_logits, indices)
    anchor_class = tf.gather_nd(anchor_class, indices)

    """
    return classification accuracy here
        ground truth = anchor_class,
        prediction   = rpn_class_logits,
    """

    return KMetrics.sparse_categorical_accuracy( tf.cast(anchor_class, "float32"), tf.cast(rpn_class_logits, "float32"))
```

Is this correct and how about calculating the accuracy for the bounding box estimation? Any ideas?

**Comment by darrell-rg** [comment]
1.  Try setting verbose =2 in model.py in call to self.keras_model.fit_generator()

2. These are the built-in losses I get from the above code:

`
Epoch 7/30
2018-07-20 23:40:35 PST- 79s - loss: 0.2756 - rpn_class_loss: 0.0095 - rpn_bbox_loss: 0.0618 - mrcnn_class_loss: 0.0837 - mrcnn_bbox_loss: 0.0464 - mrcnn_mask_loss: 0.0741 - val_loss: 0.3037 - val_rpn_class_loss: 0.0142 - val_rpn_bbox_loss: 0.0769 - val_mrcnn_class_loss: 0.0882 - val_mrcnn_bbox_loss: 0.0524 - val_mrcnn_mask_loss: 0.0719
2018-07-20 23:40:36 PST{"metric": "lr", "value": 0.00017797850887291133}
2018-07-20 23:40:36 PST{"metric": "val_loss", "value": 0.30370291173458097}
2018-07-20 23:40:36 PST{"metric": "val_rpn_class_loss", "value": 0.014248335408046842}
2018-07-20 23:40:36 PST{"metric": "val_rpn_bbox_loss", "value": 0.07692088149487972}
2018-07-20 23:40:36 PST{"metric": "val_mrcnn_class_loss", "value": 0.08822297230362892}
2018-07-20 23:40:36 PST{"metric": "val_mrcnn_bbox_loss", "value": 0.05237169526517391}
2018-07-20 23:40:36 PST{"metric": "val_mrcnn_mask_loss", "value": 0.07193737328052521}
2018-07-20 23:40:36 PST{"metric": "loss", "value": 0.27556772708892824}
2018-07-20 23:40:36 PST{"metric": "rpn_class_loss", "value": 0.009507798762060702}
2018-07-20 23:40:36 PST{"metric": "rpn_bbox_loss", "value": 0.06179905354976654}
2018-07-20 23:40:36 PST{"metric": "mrcnn_class_loss", "value": 0.08373923227190971}
2018-07-20 23:40:36 PST{"metric": "mrcnn_bbox_loss", "value": 0.04643290013074875}
2018-07-20 23:40:36 PST{"metric": "mrcnn_mask_loss", "value": 0.07408708661794662}
2018-07-20 23:40:36 PSTEpoch 8/30
`

**Comment by ApoorvaOjha** [comment]
> @darrell-rg : Where is it calculated in model ? I want the individual accuracies associated for each loss. I am currently calculating the rpn_class_accuracy and mcrnn_class_accuracy as follows:
> 
> ```
> def rpn_class_accuracy_graph(rpn_match, rpn_class_logits):
>     """RPN anchor classifier accuracy.
> 
>     rpn_match: [batch, anchors, 1]. Anchor match type. 1=positive,
>                -1=negative, 0=neutral anchor.
>     rpn_class_logits: [batch, anchors, 2]. RPN classifier logits for FG/BG.
>     """
>     # Squeeze last dim to simplify
>     rpn_match = tf.squeeze(rpn_match, -1)
>     # Get anchor classes. Convert the -1/+1 match to 0/1 values.
>     anchor_class = K.cast(K.equal(rpn_match, 1), tf.int32)
>     # Positive and Negative anchors contribute to the loss,
>     # but neutral anchors (match value = 0) don't.
>     indices = tf.where(K.not_equal(rpn_match, 0))
>     # Pick rows that contribute to the loss and filter out the rest.
>     rpn_class_logits = tf.gather_nd(rpn_class_logits, indices)
>     anchor_class = tf.gather_nd(anchor_class, indices)
> 
>     """
>     return classification accuracy here
>         ground truth = anchor_class,
>         prediction   = rpn_class_logits,
>     """
> 
>     return KMetrics.sparse_categorical_accuracy( tf.cast(anchor_class, "float32"), tf.cast(rpn_class_logits, "float32"))
> ```
> 
> Is this correct and how about calculating the accuracy for the bounding box estimation? Any ideas?

hey were you able to get clarity? is this the correct way?
